platform/security/certificate/security-crl-generator/src/main/java/org/codice/ddf/security/crl/generator/CrlGenerator.java

```diff
@@ -0,0 +1,355 @@
+      if (fileBackedOutputStream != null) {
+        try {
+          fileBackedOutputStream.reset();
+        } catch (IOException e) {
+          LOGGER.warn("Error occurred while deleting the temporary file. {}", e.getMessage());
+        }
+      }
+    }
+  }
+  /**
+   * Get the CRL from the given URL.
+   *
+   * @return - the response body
+   */
+  private Object getRemoteCrl() {
+    SecureCxfClientFactory cxfClientFactory =
+        factory.getSecureCxfClientFactory(crlLocationUrl, WebClient.class);
+    WebClient client = cxfClientFactory.getWebClient();
+    Response response = client.get();
+    return response.getEntity();
+  }
+  /**
+   * Determines the CRL encoding and creates a CRL file.
+   *
+   * @param byteSource - CRL content as a byte source
+   * @return - the created file
+   * @throws IOException
+   */
+  private File getCrlFile(ByteSource byteSource) throws IOException {
+    // Determine the file extension
+    File crlFile;
+    try (InputStream inputStream = byteSource.slice(0, 300).openStream()) {
+      if (IOUtils.toString(inputStream, StandardCharsets.UTF_8).contains("-----BEGIN")) {
+        crlFile = new File(crlFileLocation + PEM_CRL);
+      } else {
+        crlFile = new File(crlFileLocation + DEM_CRL);
+      }
+    }
+    return crlFile;
+  }
+  /**
+   * Validates the given CRL by attempting to create a {@link CRL}
+   *
+   * @param byteSource - CRL byte source
+   * @return - True if the CRL is valid. False if its invalid
+   */
+  private boolean crlIsValid(ByteSource byteSource) {
+    try (InputStream inputStream = byteSource.openStream()) {
+      CertificateFactory certificateFactory = CertificateFactory.getInstance("X.509");
+      certificateFactory.generateCRL(inputStream);
+    } catch (CertificateException | CRLException | IOException e) {
+      LOGGER.warn("An error occurred while validating the CRL. {}", e.getMessage());
+      return false;
+    }
+    return true;
+  }
+  /**
+   * Writes out the CRL to the given file and adds its path to the property files.
+   *
+   * @param byteSource - CRL byte source
+   * @param crlFile - the file to write the CRL to
+   */
+  private void writeCrlToFile(ByteSource byteSource, File crlFile) {
+    try {
+      AccessController.doPrivileged(
+          (PrivilegedExceptionAction<Void>)
+              () -> {
+                // Write out to file
+                try (OutputStream outStream = new FileOutputStream(crlFile);
+                    InputStream inputStream = byteSource.openStream()) {
+                  IOUtils.copy(inputStream, outStream);
+                  SecurityLogger.audit(
+                      "Copied the content of the CRl at {} to the local CRL at {}.",
+                      crlLocationUrl,
+                      crlFile.getPath());
+                  setCrlFileLocationInPropertiesFile(crlFile.getPath());
+                }
+                return null;
+              });
+    } catch (PrivilegedActionException e) {
```

Comment:
❗️ Same thing about wrapped `InterruptedIOException` as described above. They should be left to propagate up.

Reply:
InterruptedIOException will not be thrown here

Reply:
If you call `cancel(true)` on the future, the code invoke by the `doPrivileged()` will throw `InterruptedIOException` which will be wrapped into the `PrivilegedActionException` and therefore should be handled appropriately by checking its cause to see if it is an `InterruptedIOException`